src/hera/shared/_type_util.py

```diff
@@ -82,6 +83,13 @@ def get_workflow_annotation(annotation: Any) -> Optional[Union[Artifact, Paramet
     return metadata[0]
 
 
+def add_metadata_from_type(parameter: Parameter, annotation: Any) -> None:
```

Comment:
should be great if docstring is added here

Reply:
are there any other things you want to do in this function? this function name looks generic.

Reply:
That was the intention, but looking at the Parameter type, I can't see any other fields that we would want to set based on the type annotation, so renamed to be more specific -- thanks!